Assets/Plugins/HEBGraph/Scripts/CurveText.cs

```diff
@@ -0,0 +1,138 @@
+
+//MIT License
+
+//Copyright(c) 2019 Antony Vitillo(a.k.a. "Skarredghost")
+
+//Permission is hereby granted, free of charge, to any person obtaining a copy
+//of this software and associated documentation files (the "Software"), to deal
+//in the Software without restriction, including without limitation the rights
+//to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//copies of the Software, and to permit persons to whom the Software is
+//furnished to do so, subject to the following conditions:
+
+//The above copyright notice and this permission notice shall be included in all
+//copies or substantial portions of the Software.
+
+//THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//SOFTWARE.
+
+//Code inspired by the one of the WarpTextExample of TextMeshPro package
+
+using UnityEngine;
+using System.Collections;
+using TMPro;
+using System;
+
+namespace HEBGraph
+{
+	/// <summary>
+	/// Base class for drawing a Text Pro text following a particular curve
+	/// </summary>
+	public abstract class CurveText : MonoBehaviour
+	{
+		private TMP_Text m_TextComponent;
+		bool m_dirty = true;
+
+		private void Awake()
+		{
+			m_TextComponent = gameObject.GetComponent<TMP_Text>();
+		}
+
+		//private void OnEnable()
```

Comment:
Commented code, as always see if it makes sense to leave it there for the future, otherwise remove it! :)

Reply:
same at line 54